apps/i2c-accelerometer-gyroscope/src/i2c-accelerometer-gyroscope.c

```diff
@@ -0,0 +1,179 @@
+#define MPU6050_GYRO_REG_DATA_START		0X43
+
+/* Full Scale Range to +- 2g */
+#define FS_SEL					0x00
+/* Full Scal Range to 250deg/seg */
+#define AFS_SEL					0x00
+
+/* Time interval (in mseg) to get new measures */
+#define TIME_INTERVAL				200
+
+#define REGISTERS_NO				6
+#define AXIS_NO					3
+#define SENSORS_NO				6
+
+#define X_AXIS					0
+#define Y_AXIS					1
+#define Z_AXIS					2
+
+/*
+ * Note that measured data has 16 bits, what makes necessary to put together
+ * the MSB's with their LSB's.
+ * Example:
+ * buf[0] will have MSB's for x axis acceleration and buf[1] will have LSB's
+ * for x axis acceleration, that is, the raw data of x axis acceleration will
+ * be buf16[0] = (buf[0] << 8) | buf(1)
+ */
+#define MERGE_MSB_LSB(a,b) ( a << 8 | b )
+
+/* Buffer vector to store data from sensor 8 bit registers */
+u8_t accel_reg_buffer[REGISTERS_NO], gyro_reg_buffer[REGISTERS_NO];
+/* Buffer vector to store measured data from sensor */
+s16_t accel_measurements[AXIS_NO], gyro_measurements[AXIS_NO];
+
+/*
+ * Each vector position corresponds to an axis,
+ * e.g.:
+ * accel[0] -> acceleration in x axis
+ * accel[1] -> acceleration in y axis
+ * accel[2] -> acceleration in z axis
+ * the same logic stands for gyroscope measurements
+ */
+s32_t accel[AXIS_NO], gyro[AXIS_NO];
+
+/*
+ * Offset values for sensor calibration
+ * To obtain these values, put the sensor in your chosen origin and realise a
+ * group of measurements, then take the mean value for each variable.
+ * Obs: Note that if you're using a high precision sensibility, the measures
+ * will vary slightly everytime.
+ */
+s32_t accel_offset[AXIS_NO] = { 236,
+				-752,
+				20996
+				};
+s32_t gyro_offset[AXIS_NO] = {  75,
+				-147,
+				23
+				};
+
+/* Name list vector for KNoT sensors */
+char *name_list[SENSORS_NO] = { "ACCELX", "ACCELY", "ACCELZ",
+				"GYROX", "GYROY", "GYROZ" };
+
+LOG_MODULE_REGISTER(accelgyro, LOG_LEVEL_DBG);
+
+struct device *i2c_dev;
+
+void setup(void)
+{
+	/* Peripherals control */
+	i2c_dev = device_get_binding(DT_NORDIC_NRF_I2C_0_LABEL);
+	/* Checks if peripheral was properly configured */
+	if (!i2c_dev) {
+		printk("I2C: Device driver not found.\n");
+
+		return;
+	}
+	/* Take sensor out of sleep mode */
+	i2c_reg_update_byte(i2c_dev, MPU_ADDRESS,
+				MPU6050_REG_PWR_MGMT1, MPU6050_SLEEP_EN, 0);
+	/* Set the full scale range for gyroscope */
+	i2c_reg_write_byte(i2c_dev, MPU_ADDRESS, MPU6050_REG_GYRO_CFG, FS_SEL);
+	/* Set the full scale range for accelerometer */
+	i2c_reg_write_byte(i2c_dev, MPU_ADDRESS,
+				MPU6050_REG_ACCEL_CFG, AFS_SEL);
+	/* KNoT config */
+	int i = 0;
+	for (i = 0; i < AXIS_NO; i++) {
+		knot_data_register(i, name_list[i], KNOT_TYPE_ID_ACCELERATION,
+					KNOT_VALUE_TYPE_INT,
+					KNOT_UNIT_ACCELERATION_MS2,
+					&accel[i], sizeof(accel[i]),
+					NULL, NULL);
+		knot_data_config(i, KNOT_EVT_FLAG_CHANGE, NULL);
+	}
+	for (i = 0; i < AXIS_NO; i++) {
+		knot_data_register(i+3, name_list[i+3],
```

Comment:
Add spaces between the operator '+' and create a define for 3.

Reply:
Done